infoschema/builder.go

```diff
@@ -89,6 +86,15 @@ func (b *Builder) ApplyDiff(m *meta.Meta, diff *model.SchemaDiff) ([]int64, erro
 	}
 	dbInfo := b.copySchemaTables(roDBInfo.Name.L)
 	b.copySortedTables(oldTableID, newTableID)
+	switch diff.Type {
+	case model.ActionAlterTableAlterPartition:
+		_, err := b.applyCreateTable(m, dbInfo, diff.TableID, nil, model.ActionAlterTableAlterPartition, nil)
+		if err != nil {
+			return nil, err
+		}
+		// TODO: enhancement: If the leader Placement Policy isn't updated, maybe we can omit the diff.
+		return []int64{diff.PartitionID}, b.applyPlacementUpdate(placement.GroupID(diff.PartitionID))
+	}
```

Comment:
`applyDropTable` should be called before `applyCreateTable`, otherwise `b.is.sortedTablesBuckets` and `dbInfo.Tables` has duplicate tables.
So I suggest to follow the process of `ActionTruncateTablePartition`.

Reply:
addressed.

Reply:
No, I mean:
1. remove this block.
2. fill the partition id in `diff.AffectedOpts`.
3. add a judgment in `switch diff.Type`, just like what `ActionTruncateTablePartition` does.
It's more natural.

Reply:
addressd.